pages/learn/develop/dockerfile.md

```diff
@@ -101,6 +101,20 @@ Here are the supported machine names and architectures:
 
 {{> "general/deviceTypeNames"}}
 
+## Multiple Dockerfiles
+
+There are cases when you would need a higher granularity of control when specifying build instructions for different devices and architectures than a Dockerfile template can provide. An example of this would be when different configuration or installation files are required.
+
+When deploying an application, the balena CLI tool looks at all available dockerfiles and builds the appropriate docker image based on the following order:
+
+* Dockerfile.<device-type>
+* Dockerfile.<arch>
+* Dockerfile.template
+
+As an example, if you have two dockerfiles in the directory, `Dockerfile.raspberrypi3` and `Dockerfile.template`, whenever you publish the application do balenaCloud, if the `device-type` is a Raspberry Pi 3, `Dockerfile.raspberrypi3` will be selected and for all other devices it will automatically select `Dockerfile.template`.
```

Comment:
"do balenaCloud" → "to balenaCloud"
The sentence got a bit long, maybe break it in too:
> As an example, say there are two dockerfiles in the directory, `Dockerfile.raspberrypi3` and `Dockerfile.template`. When the app is deployed to balenaCloud, if the `device-type` is a Raspberry Pi 3, `Dockerfile.raspberrypi3` will be selected and for all other devices it will automatically select `Dockerfile.template`.